templates/hello/simple-snippet.html

```diff
@@ -0,0 +1,204 @@
+            });
+          };
+
+          Mozilla.UITour.showMenu = function(name, callback) {
+            var showCallbackID;
+            if (callback)
+              showCallbackID = _waitForCallback(callback);
+
+            _sendEvent('showMenu', {
+              name: name,
+              showCallbackID: showCallbackID,
+            });
+          };
+
+          Mozilla.UITour.hideMenu = function(name) {
+            _sendEvent('hideMenu', {
+              name: name
+            });
+          };
+
+          Mozilla.UITour.getConfiguration = function(configName, callback) {
+            _sendEvent('getConfiguration', {
+              callbackID: _waitForCallback(callback),
+              configuration: configName,
+            });
+          };
+
+          Mozilla.UITour.trackHelloStatus = function () {
+            Mozilla.UITour.getConfiguration('availableTargets', function(config) {
+              if (config.targets && config.targets.indexOf('loop') > -1) {
+                sendMetric('hello-eligible-view');
+              } else {
+                sendMetric('hello-ineligible-view');
+              }
+            });
+          };
+
+          function isTargetAvailable (target, callback) {
+            Mozilla.UITour.getConfiguration('availableTargets', function(config) {
+              callback(config.targets && config.targets.indexOf(target) > -1);
+            });
+          }
+
+          function bindHelloObserver () {
+            Mozilla.UITour.observe(function(e) {
+              switch (e) {
+                case 'Loop:ChatWindowOpened':
+                  sendMetric('hello-start-conversation');
+                  break;
+                case 'Loop:RoomURLCopied':
+                  sendMetric('hello-url-copied');
+                  break;
+                // bug 1118346
+                case 'Loop:RoomURLEmailed':
+                  sendMetric('hello-url-emailed');
+                  break;
+              }
+            });
+          }
+
+         function handleVisibilityChange () {
+            if (document.hidden) {
+              // hide Hello menu & stop observer when changing tabs or minimizing window
+              Mozilla.UITour.observe(null);
+            } else {
+              // listen for Hello menu/chat window events
+              bindHelloObserver();
+            }
+          }
+
+          function thumbsUp () {
+            bindHelloObserver();
+            Mozilla.UITour.showMenu ('loop');
+            // bug 1115227
+            Mozilla.UITour.registerPageID('hello-tour_OpenPanel_snippet-{{ snippet_id }}');
+            // bug 1113896
+            document.addEventListener('click', function(e) {
+                Mozilla.UITour.hideMenu ('loop');
+            }, false);
+            document.addEventListener('visibilitychange', handleVisibilityChange);
+            sendMetric('hello-open');
+          };
+
+          var buttons = document.getElementsByClassName('launch-hello');
```

Comment:
A few things:

1. This selects all buttons with a `launch-hello` classname in the entire document, including other instances of this snippet. You should use the `snippet` DOM element you have above to limit your search to only select elements within this individual snippet.

2. This code executes as soon as the snippet is embedded in the DOM. It's preferred to not execute anything unless this snippet is the one being shown, so this block of code should be moved to the inside of the `show_snippet` event handler.